### PR TITLE
[ru] cs-221-variables-models

### PR DESCRIPTION
**Pull request by AlexandrParkhomenko**
web format: [Variables-based models](https://alexandrparkhomenko.github.io/stanford/cs-221/variables-models/index.html)

### CODE REVIEW

File: ru/cs-221-variables-models.md [hunk 1]
```diff
@@ -0,0 +1,616 @@
...
+**30. [Gibbs sampling ― Gibbs sampling is an iterative approximate method that modifies the assignment of a factor graph one variable at a time until convergence. At step i:, we assign to each element u∈Domaini a weight w(u) that is the product of all factors connected to that variable, we sample v from the probability distribution induced by w and assign it to Xi.]**
+
+&#10230; [Выборка по Гиббсу ― это итеративный приближенный метод; который изменяет присвоение факторному графу одной переменной за раз до сходимости. На шаге i:, мы присваиваем каждому элементу u∈Domaini вес w(u). Вес является произведением всех связанных с этой переменной факторов, мы выбираем v из индуцированного w распределения вероятностей и присваиваем его Xi.]
+
+<br>
+
+
+**31. Remark: Gibbs sampling can be seen as the probabilistic counterpart of ICM. It has the advantage to be able to escape local minima in most cases.**
+
+&#10230; Примечание: Выборка по Гиббсу можно рассматривать как вероятностный аналог ICM. Преимущество этого метода заключается в возможности избежать локальных минимумов в большинстве случаев.
```
Comment (geotrush) [inline ru/cs-221-variables-models.md:217]:
```suggestion
&#10230; Примечание: Выборку по Гиббсу можно рассматривать как вероятностный аналог ICM. Преимущество этого метода заключается в возможности избежать локальных минимумов в большинстве случаев.
```

File: ru/cs-221-variables-models.md [hunk 2]
```diff
@@ -0,0 +1,616 @@
...
+**34. Remark: independence is the key property that allows us to solve subproblems in parallel.**
+
+&#10230; Примечание: независимость - ключевое свойство, которое позволяет нам решать подзадачи параллельно.
+
+<br>
+
+
+**35. Conditional independence ― We say that A and B are conditionally independent given C if conditioning on C produces a graph in which A and B are independent. In this case, it is written:**
+
+&#10230; Условная независимость ― Мы говорим, что A и B условно независимы для данного C, если кондиционирование на C дает граф, в котором A и B независимы. В этом случае пишется:
```
Comment (geotrush) [inline ru/cs-221-variables-models.md:245]:
```suggestion
&#10230; Условная независимость ― Мы говорим, что A и B условно независимы для данного C, если принятое условие C дает граф, в котором A и B независимы. В этом случае пишется:
```

File: ru/cs-221-variables-models.md [hunk 3]
```diff
@@ -0,0 +1,616 @@
...
+**43. Bayesian networks**
+
+&#10230; Байесовские сети
+
+<br>
+
+
+**44. In this section, our goal will be to compute conditional probabilities. What is the probability of a query given evidence?**
+
+&#10230; В этом разделе нашей целью будет вычисление условных вероятностей. Какова вероятность того, что запрос предоставит доказательства?
```
Comment (geotrush) [inline ru/cs-221-variables-models.md:308]:
```suggestion
&#10230; В этом разделе нашей целью будет вычисление условных вероятностей. Какова условная вероятность запроса при данных наблюдениях?
```

File: ru/cs-221-variables-models.md [hunk 4]
```diff
@@ -0,0 +1,616 @@
...
+**47. Directed acyclic graph ― A directed acyclic graph (DAG) is a finite directed graph with no directed cycles.**
+
+&#10230; Направленный ациклический граф ― Directed acyclic graph (DAG) - это конечный ориентированный граф без ориентированных циклов.
+
+<br>
+
+
+**48. Bayesian network ― A Bayesian network is a directed acyclic graph (DAG) that specifies a joint distribution over random variables X=(X1,...,Xn) as a product of local conditional distributions, one for each node:**
+
+&#10230; Байесовская сеть ― Байесовская сеть - это ориентированный ациклический граф (DAG), который определяет совместное распределение по случайным величинам X=(X1,...,Xn) как произведение локальных условных распределений, по одному для каждого узла:
```
Comment (geotrush) [inline ru/cs-221-variables-models.md:336]:
```suggestion
&#10230; Байесовская сеть ― Bayesian network - это ориентированный ациклический граф (DAG), который определяет совместное распределение по случайным величинам X=(X1,...,Xn) как произведение локальных условных распределений, по одному для каждого узла:
```

File: ru/cs-221-variables-models.md [hunk 5]
```diff
@@ -0,0 +1,616 @@
...
+**54. Probabilistic programs**
+
+&#10230; Вероятностные программы
+
+<br>
+
+
+**55. Concept ― A probabilistic program randomizes variables assignment. That way, we can write down complex Bayesian networks that generate assignments without us having to explicitly specify associated probabilities.**
+
+&#10230; Концепция ― вероятностная программа рандомизирует присвоение переменных. Таким образом, мы можем записывать сложные байесовские сети, которые генерируют назначения, без необходимости явно указывать связанные вероятности.
```
Comment (geotrush) [inline ru/cs-221-variables-models.md:385]:
```suggestion
&#10230; Концепция ― вероятностная программа случайным образом присваивает переменные. Таким образом, мы можем записывать сложные байесовские сети, которые генерируют назначения, без необходимости явно указывать связанные вероятности.
```

File: ru/cs-221-variables-models.md [hunk 6]
```diff
@@ -0,0 +1,616 @@
...
+**62. Inference**
+
+&#10230; Байесовский вывод
+
+<br>
+
+
+**63. [General probabilistic inference strategy ― The strategy to compute the probability P(Q|E=e) of query Q given evidence E=e is as follows:, Step 1: Remove variables that are not ancestors of the query Q or the evidence E by marginalization, Step 2: Convert Bayesian network to factor graph, Step 3: Condition on the evidence E=e, Step 4: Remove nodes disconnected from the query Q by marginalization, Step 5: Run a probabilistic inference algorithm (manual, variable elimination, Gibbs sampling, particle filtering)]**
+
+&#10230; [Общая стратегия вероятностного вывода ― Стратегия вычисления вероятности P(Q|E=e) запроса Q при наличии свидетельства (англ: evidence) E=e следующая:, Шаг 1. Удалить неявляющиеся предками запроса Q или свидетельства E переменные путем маргинализации, Шаг 2: Преобразовать байесовскую сеть в факторный граф, Шаг 3: Улучшить состояние свидетельства E=e, Шаг 4: Удалить отключенные от запроса Q узлы путем маргинализации, Шаг 5: Запустить вероятностный алгоритм вывода (вручную, исключение переменных, выборка по Гиббсу, фильтрация частиц)]
```
Comment (geotrush) [inline ru/cs-221-variables-models.md:441]:
```suggestion
&#10230; [Общая стратегия вероятностного вывода ― Стратегия вычисления вероятности P(Q|E=e) запроса Q при наличии наблюдаемых значений E=e следующая:, Шаг 1. Удалить не являющиеся предками запроса Q или наблюдения E переменные путем маргинализации, Шаг 2: Преобразовать байесовскую сеть в факторный граф, Шаг 3: Улучшить состояние наблюдения E=e, Шаг 4: Удалить отключенные от запроса Q узлы путем маргинализации, Шаг 5: Запустить вероятностный алгоритм вывода (вручную, исключение переменных, выборка по Гиббсу, фильтрация частиц)]
```

File: ru/cs-221-variables-models.md [hunk 7]
```diff
@@ -0,0 +1,616 @@
...
+**67. Remark: this algorithm interprets each assignment to be a path where each edge hi−1→hi is of weight p(hi|hi−1)p(ei|hi).**
+
+&#10230; Примечание: этот алгоритм интерпретирует каждое присвоение как путь, в котором каждое ребро hi−1→hi имеет вес p(hi|hi−1)p(ei|hi).
+
+<br>
+
+
+**68. [Gibbs sampling ― This algorithm is an iterative approximate method that uses a small set of assignments (particles) to represent a large probability distribution. From a random assignment x, Gibbs sampling performs the following steps for i∈{1,...,n} until convergence:, For all u∈Domaini, compute the weight w(u) of assignment x where Xi=u, Sample v from the probability distribution induced by w: v∼P(Xi=v|X−i=x−i), Set Xi=v]**
+
+&#10230; [Выборка по Гиббсу ― Этот алгоритм представляет собой итеративный приближенный метод. Он использует небольшой набор назначений (частиц) для представления большого распределения вероятностей. Из случайного присвоения x, Выборка по Гиббсу выполняет следующие шаги для i∈{1,...,n} до сходимости:, Для всех u∈Domaini, вычислить вес w(u) присвоения x; где Xi=u, Произвести выборку v из индуцированного w распределения вероятностей: v∼P(Xi=v|X−i=x−i), Установить Xi=v]
```
Comment (geotrush) [inline ru/cs-221-variables-models.md:476]:
```suggestion
&#10230; [Выборка по Гиббсу ― Этот алгоритм представляет собой итеративный приближенный метод. Он использует небольшой набор присвоений (частиц) для представления большого распределения вероятностей. Из случайного присвоения x, Выборка по Гиббсу выполняет следующие шаги для i∈{1,...,n} до сходимости:, Для всех u∈Domaini, вычислить вес w(u) присвоения x; где Xi=u, Произвести выборку v из индуцированного w распределения вероятностей: v∼P(Xi=v|X−i=x−i), Установить Xi=v]
```

File: ru/cs-221-variables-models.md [hunk 8]
```diff
@@ -0,0 +1,616 @@
...
+
+&#10230; [Фильтрация частиц ― Этот алгоритм аппроксимирует апостериорную плотность переменных состояния с учетом данных наблюдений путем отслеживания K частиц за раз. Начиная с набора частиц C размера K, мы итеративно выполняем следующие 3 шага:, Шаг 1: предложение - Для каждой старой частицы xt−1∈C, выбрать x из распределения вероятности перехода p(x|xt−1) и добавить x к множеству C′., Шаг 2: взвешивание - Взвесить каждый x из множества C′ как w(x)=p(et|x), где et - свидетельство в момент времени t., Шаг 3: повторная выборка - выборка K элементов из множества C′ с использованием индуцированного w распределения вероятностей и сохранение их в C: это текущие частицы xt.]
```
Comment (geotrush) [inline ru/cs-221-variables-models.md:490]:
```suggestion
&#10230; [Фильтрация частиц ― Этот алгоритм аппроксимирует апостериорную плотность переменных состояния с учетом данных наблюдений путем отслеживания K частиц за раз. Начиная с набора частиц C размера K, мы итеративно выполняем следующие 3 шага:, Шаг 1: предложение - Для каждой старой частицы xt−1∈C, выбрать x из распределения вероятности перехода p(x|xt−1) и добавить x к множеству C′., Шаг 2: взвешивание - Взвесить каждый x из множества C′ как w(x)=p(et|x), где et - наблюдение в момент времени t., Шаг 3: повторная выборка - выборка K элементов из множества C′ с использованием индуцированного w распределения вероятностей и сохранение их в C: это текущие частицы xt.]
```

File: ru/cs-221-variables-models.md [hunk 9]
```diff
@@ -0,0 +1,616 @@
...
+**74. Algorithm ― The Expectation-Maximization (EM) algorithm gives an efficient method at estimating the parameter θ through maximum likelihood estimation by repeatedly constructing a lower-bound on the likelihood (E-step) and optimizing that lower bound (M-step) as follows:**
+
+&#10230; Алгоритм Expectation-Maximization (EM) ― алгоритм максимизации ожидания дает эффективный метод оценки параметра θ посредством оценки максимального правдоподобия путем многократного построения нижней границы правдоподобия (E-шаг) и оптимизации этой нижней границы (M-шаг) следующим образом:
+
+<br>
+
+
+**75. [E-step: Evaluate the posterior probability q(h) that each data point e came from a particular cluster h as follows:, M-step: Use the posterior probabilities q(h) as cluster specific weights on data points e to determine θ through maximum likelihood.]**
+
+&#10230; [Шаг E: Оценить апостериорную вероятность q(h) прихода каждой точки данных e из определенного кластера h следующим образом:, M-шаг: использовать апостериорные вероятности q(h) в качестве весовых коэффициентов для конкретных кластеров точек данных e и определить θ через максимальное правдоподобие.]
```
Comment (geotrush) [inline ru/cs-221-variables-models.md:525]:
```suggestion
&#10230; [Шаг E: Оценить апостериорную вероятность q(h) принадлежности каждой точки данных e из определенного кластера h следующим образом:, M-шаг: использовать апостериорные вероятности q(h) в качестве весовых коэффициентов для конкретных кластеров точек данных e и определить θ через максимальное правдоподобие.]
```

File: ru/cs-221-variables-models.md [hunk 10]
```diff
@@ -0,0 +1,616 @@
...
+**83. View PDF version on GitHub**
+
+&#10230; Посмотреть PDF-версию на GitHub
+
+<br>
+
+
+**84. Original authors**
+
+&#10230; Авторы оригинала: Afshine Amidi и Shervine Amidi ― https://github.com/shervinea
```
Comment (geotrush) [inline ru/cs-221-variables-models.md:588]:
```suggestion
&#10230; Авторы оригинала: Afshine Amidi и Shervine Amidi ― https://github.com/afshinea и https://github.com/shervinea
```

File: ru/cs-221-variables-models.md [hunk 11]
```diff
@@ -0,0 +1,616 @@
...
+
+<br>
+
+
+**85. Translated by X, Y and Z**
+
+&#10230; Российская адаптация: Пархоменко Александр ― https://github.com/AlexandrParkhomenko
```
Comment (geotrush) [inline ru/cs-221-variables-models.md:595]:
```suggestion
&#10230; Переведено на русский язык: Пархоменко Александр ― https://github.com/AlexandrParkhomenko
```

File: ru/cs-221-variables-models.md [hunk 12]
```diff
@@ -0,0 +1,616 @@
...
+
+<br>
+
+
+**86. Reviewed by X, Y and Z**
+
+&#10230; Проверено X, Y и Z
```
Comment (geotrush) [inline ru/cs-221-variables-models.md:602]:
```suggestion
&#10230; Проверено на русском языке: Труш Георгий (Georgy Trush) ― https://github.com/geotrush
```

File: ru/cs-221-variables-models.md [hunk 13]
```diff
@@ -0,0 +1,616 @@
...
+**87. By X and Y**
+
+&#10230; По X и Y
+
+<br>
+
+
+**88. The Artificial Intelligence cheatsheets are now available in [target language].**
+
+&#10230; Шпаргалки по искусственному интеллекту теперь доступны в формате [target language].
```
Comment (geotrush) [inline ru/cs-221-variables-models.md:616]:
```suggestion
&#10230; Шпаргалки по искусственному интеллекту теперь доступны на русском языке.
```

**Review by tensorush — CHANGES_REQUESTED** [review]
This one is especially good! I've pointed out some minor typos.

**Pull request closed without merging**